### ISSUE TITLE
Body / Page scroll happening when clicking on dropdown menu

### ISSUE THREAD

**Issue by ROJAYADHAV** [vmware-clarity/ng-clarity#412]
Body / Page scroll happening when clicking on dropdown menu

## Describe the bug

Main page scrolling when click on dropdown menu

## How to reproduce

Angular v - 13.3.11
 "@cds/core": "^6.2.2",
    "@clr/angular": "~13.10.0",
    "@clr/icons": "~13.0.2",
    "@clr/ui": "~13.10.0",
node v - 14.17.5

<clr-dropdown >
  <button [disabled]="isDisabled" [ngClass]="{'sct-btn-disabled': isDisabled}" class="btn dropdown-button"
          clrDropdownTrigger [tabindex]="field.tabindex" #dropdownButton>
    {{ currentOption.label}}
    <clr-icon *ngIf="!dropdownOpen" shape="caret down" class="icon-orient-down"></clr-icon>
    <clr-icon *ngIf="dropdownOpen" shape="caret up" class="icon-orient-up"></clr-icon>
  </button>
  <ng-template [clrIfOpen]="dropdownOpen" (clrIfOpenChange)="toggleOpen($event)">
    <clr-dropdown-menu class="sct-dropdown-menu" [tabIndex]="-1" [clrPosition]="menuPosition">
      <div >{{option.label}}</div>
 </clr-dropdown-menu>
  </ng-template>
</clr-dropdown>

This is the code we are currently using . When we click the dropdown menu , the whole page is scrolling to top and also to bottom sometimes. Can anyone please help me for this.

**Comment by ROJAYADHAV** [comment]
<clr-dropdown >
  <button [disabled]="isDisabled" [ngClass]="{'sct-btn-disabled': isDisabled}" class="btn dropdown-button"
          clrDropdownTrigger [tabindex]="field.tabindex" #dropdownButton>
    {{ currentOption.label}}
    <clr-icon *ngIf="!dropdownOpen" shape="caret down" class="icon-orient-down"></clr-icon>
    <clr-icon *ngIf="dropdownOpen" shape="caret up" class="icon-orient-up"></clr-icon>
  </button>
  <ng-template [clrIfOpen]="dropdownOpen" (clrIfOpenChange)="toggleOpen($event)">
    <clr-dropdown-menu class="sct-dropdown-menu" [tabIndex]="-1" [clrPosition]="menuPosition">
      <div >{{option.label}}</div>
 </clr-dropdown-menu>
  </ng-template>
</clr-dropdown>


this is the template

**Comment by Jinnie** [comment]
Please, create a StackBlitz reproduction of the issue. These are our starter templates:
https://stackblitz.com/@clr-team/

**Comment by ROJAYADHAV** [comment]
Please find the below link where on click of last 2 dropdowns I can see window is getting scrolled.

https://stackblitz.com/edit/clarity-light-theme-clr13-cds6-abht4s?file=src/app/app.component.html

**Comment by Jinnie** [comment]
Is it fully saved? 
I only see several dropdowns and no scrolling container. If I add some white space above them, and then start clicking on the dropdowns, I can't reproduce scroll to bottom/top. It only adds some space at the end of the document.
I've heard similar problems reported before, though, so I want to make sure we properly reproduce it.
Could you describe step by step exactly what you are doing and what are you observing? A screen cast may also help.

**Comment by ROJAYADHAV** [comment]
It is fully saved @Jinnie. Can you please try to run this example in new tab. 

step 1: Click on last dropdown down arrow. After opening the last dropdown menu there is a vertical scroll movement.

Please find the snapshot below where I have highlighted the scroller issue.

![image](https://user-images.githubusercontent.com/25585195/205648602-9eb7e1f4-7a5c-43eb-b3f8-6b6aaa7ff0b2.png)

**Comment by kevinbuhmann** [comment]
It seems to be scrolling the dropdown into view which I think should be expected.

**Comment by ROJAYADHAV** [comment]
Hi @kevinbuhmann I can see this behavior is not happening in case of angular 9 and clarity 3.x version

**Comment by kevinbuhmann** [comment]
> Hi @kevinbuhmann I can see this behavior is not happening in case of angular 9 and clarity 3.x version

It could be a new feature. Maybe @Jinne knows.

**Comment by ROJAYADHAV** [comment]
> > Hi @kevinbuhmann I can see this behavior is not happening in case of angular 9 and clarity 3.x version
> 
> It could be a new feature. Maybe @Jinne knows.

Hi @Jinnie Can you please confirm regarding the same.

**Comment by ROJAYADHAV** [comment]
One more observation that I have seen is when I am using <clr-dropdown> inside <clr-dropdown-menu> I can see that parent/document window vertical scroll is not happening. But it gave us another issue regarding keyboard up/down events where we cannot navigate through dropdown menu options.

**Comment by ietabhi** [comment]
I am facing same problem as well, we have dropdown content as absolute position so ideally it should not jump page. my customer will not like jumping pages specially when they click on dropdown.

**Comment by ROJAYADHAV** [comment]
Hi @Jinnie , Any updates on this issue?

**Comment by Jinnie** [comment]
What you observe as changed behavior may be related to the focus navigation, which has changed since v3, which is out of support for at least 2 years now.
I am still unable to reproduce any problem with the provided Stackblitz, unless I resize to a very small window, where some container resizing is naturally forced and leads to scroll event.

**Comment by ROJAYADHAV** [comment]
We are using chrome browser and windows 10

**Comment by Jinnie** [comment]
I can not find anything in the code that's forcing a scroll into view, but it's possible to be related to the first item focusing.
The dropdown component is using its own focus management solution. In v13 we may want to offload all focus management to the CDK utilities.
Any work on this bug should happen after the CDK is merged in our codebase. 
I'm assigning this to myself for proper tracking, when the above changes get merged.

**Comment by Jinnie** [comment]
Waiting for this PR to get merged, before we can work on changing the focus management solution:
https://github.com/vmware-clarity/ng-clarity/pull/426/
